lib/logstash/filters/aggregate.rb

```diff
@@ -597,10 +677,17 @@ def flush(options = {})
     elsif @@flush_instance_map[@task_id].timeout.nil?
       @@flush_instance_map[@task_id].timeout = @@default_timeout
     end
-    
+
+    events_to_flush = []
+    # handle inactivity timeout if set and only every @inactivity_timeout / 2 seconds
+    if !@inactivity_timeout.nil? && (@@last_flush_inactivity_timeout_timestamp.nil? || Time.now > @@last_flush_inactivity_timeout_timestamp + @inactivity_timeout / 2)
+      events_to_flush = remove_inactive_maps()
+      @@last_flush_inactivity_timeout_timestamp = Time.now
+    end
```

Comment:
I disagree with the way to do that.
timeout and inactivity_timeout management must be a common part, not separate.
In the same idea, for performance reasons, I prefer that all maps are looped once per flush() call, and not twice.
I prefer there is only one method `remove_expired_maps()` that process both timeout and inactivity_timeout.

So I invite you to :
- remove this block above
- move your code in "remove_inactive_maps()" to "remove_expired_maps()"
- then to avoid NullPointerException, add the following code :
``` ruby
     if @@flush_instance_map[@task_id].inactivity_timeout.nil?
       @@flush_instance_map[@task_id].inactivity_timeout = @@flush_instance_map[@task_id].timeout
     end
```
- and to finish, update the `if` below :
```
# Launch timeout management only every interval of (@inactivity_timeout / 2) seconds or at Logstash shutdown
if @@flush_instance_map[@task_id] == self && (!@@last_flush_timestamp_map.has_key?(@task_id) || Time.now > @@last_flush_timestamp_map[@task_id] + @inactivity_timeout / 2 || options[:final])
```